svnr doesn't pass the test in some cases

```console
1) Generate SVNR number
       SVNR number:

      AssertionError: expected false to be true
      + expected - actual

      -false
      +true

      at Context.<anonymous> (test/svnr.spec.ts:17:35)
```

--- a/src/svnr/check.ts
+++ b/src/svnr/check.ts
@@ -7,6 +7,7 @@
  */
 
 import CheckResult from '../CheckResult';
+import checkDigit from './checkDigit';
 
 const check = (input: string): CheckResult<{}> => {
     const v = input.replace(/\s/g, '');
@@ -14,22 +15,11 @@ const check = (input: string): CheckResult<{}> => {
         return { valid: false };
     }
 
-    const weight = [3, 7, 9, 5, 8, 4, 2, 1, 6];
-    const getCheckDigit = (str: string) => {
-        return str.split('').map((c, i) => parseInt(c, 10) * weight[i]).reduce((a, b) => a + b, 0) % 11;
-    };
+    const serial = parseInt(v.substr(0, 3), 10);
+    const dob = v.substr(4);
+    const cd = checkDigit(serial, dob);
 
-    let sequence = parseInt(v.substring(0, 3), 10);
-    const cd = parseInt(v.substring(3, 4), 10);
-    const dob = v.substring(4);
-
-    let checkDigit = getCheckDigit(`${sequence}${dob}`);
-    while (checkDigit === 10) {
-        sequence += 1;
-        checkDigit = getCheckDigit(`${sequence}${dob}`);
-    }
-
-    return { valid: checkDigit === cd };
+    return { valid: `${cd.checkDigit}` === v.substr(3, 1) };
 };
 
 export default check;

--- a/src/svnr/checkDigit.ts
+++ b/src/svnr/checkDigit.ts
@@ -1,0 +1,33 @@
+/**
+ * Generate fake and valid numbers
+ *
+ * @see https://fakenumbers.io
+ * @license https://fakenumbers.io/license
+ * @copyright 2020 Nguyen Huu Phuoc <me@phuoc.ng>
+ */
+
+interface CheckDigit {
+    checkDigit: number;
+    sequence: number;
+}
+
+const checkDigit = (sequence: number, suffix: string): CheckDigit => {
+    let s = sequence;
+    const weight = [3, 7, 9, 5, 8, 4, 2, 1, 6];
+    const calculateCheckDigit = (withSequence: string) => {
+        return withSequence.split('').map((c, i) => parseInt(c, 10) * weight[i]).reduce((a, b) => a + b, 0) % 11;
+    };
+
+    let cd = calculateCheckDigit(`${s}${suffix}`);
+    while (cd === 10) {
+        s += 1;
+        cd = calculateCheckDigit(`${s}${suffix}`);
+    }
+
+    return {
+        checkDigit: cd,
+        sequence: s,
+    };
+};
+
+export default checkDigit;

--- a/src/svnr/fake.ts
+++ b/src/svnr/fake.ts
@@ -8,6 +8,7 @@
 
 import randomFromRange from '../utils/randomFromRange';
 import randomNumbers from '../utils/randomNumbers';
+import checkDigit from './checkDigit';
 
 /**
  * Generate a random SVNR number (Austrian social insurance number)
@@ -17,31 +18,20 @@ import randomNumbers from '../utils/randomNumbers';
  */
 const fake = (): string => {
     // First 3 digits, can't start with zero
-    const serial = [randomFromRange(1, 9)].concat(randomNumbers(2)).join('');
+    const serial = parseInt(`${randomFromRange(1, 9)}${randomNumbers(2).join('')}`, 10);
 
     // The last 6 digits indicate the date of birth
     const month = randomFromRange(1, 12);
-    const monthWithPrefix = month > 10 ? `${month}` : `0${month}`;
+    const monthWithPrefix = month >= 10 ? `${month}` : `0${month}`;
 
     const date = randomFromRange(0, 28);
-    const dateWithPrefix = date > 10 ? `${date}` : `0${date}`;
+    const dateWithPrefix = date >= 10 ? `${date}` : `0${date}`;
 
     const year = randomNumbers(2).join('');
     const dob = `${monthWithPrefix}${dateWithPrefix}${year}`;
 
-    const weight = [3, 7, 9, 5, 8, 4, 2, 1, 6];
-    const checkDigit = (input: string) => {
-        return input.split('').map((c, i) => parseInt(c, 10) * weight[i]).reduce((a, b) => a + b, 0) % 11;
-    };
-
-    let sequence = parseInt(serial, 10);
-    let cd = checkDigit(`${serial}${dob}`);
-    while (cd === 10) {
-        sequence += 1;
-        cd = checkDigit(`${sequence}${dob}`);
-    }
-
-    return `${sequence}${cd} ${dob}`;
+    const cd = checkDigit(serial, dob);
+    return `${cd.sequence}${cd.checkDigit} ${dob}`;
 };
 
 export default fake;

--- a/test/svnr.spec.ts
+++ b/test/svnr.spec.ts
@@ -10,10 +10,22 @@ import { expect } from 'chai';
 
 import { svnr } from '../src';
 
-describe('Generate SVNR number', () => {
-    it('SVNR number', () => {
+describe('SVNR number', () => {
+    it('Fake SVNR number', () => {
         const number = svnr.fake();
         const result = svnr.check(number);
         expect(result.valid).to.be.true;
+    });
+
+    it('Valid SVNR numbers', () => {
+        [
+            '1237 010180', '4298021282', '5659081577', '8852041693', '5497042152',
+        ].forEach((n) => expect(svnr.check(n).valid).to.be.true);
+    });
+
+    it('Invalid SVNR numbers', () => {
+        [
+            '4298001282', '1508-10-13-50', '1508101051', '4290021282', '21 34 23 12 74',
+        ].forEach((n) => expect(svnr.check(n).valid).to.be.false);
     });
 });

